### ISSUE TITLE
Audio monitor output stops after extended playback

### ISSUE THREAD

**Issue by michaeltoohig** [obsproject/obs-studio#2348]
## Expected Behavior
When "Monitor and Output" is selected for an audio source the audio should always play for the monitor and output.

## Current Behavior
OBS audio channel says "Monitor and Output"; however, audio monitor does not work after extended use of days.  Which requires manually setting the audio monitoring option to "Monitor Off" then back to "Monitor and Output" for the audio to return to monitor.

## Steps to Reproduce
1. Play a Media Source on a scene in a projector
2. Change the video periodically via websockets
3. Repeat for days
4. Eventually audio monitor fails

## Additional information
 I am using websockets to control the media source on the scene which changes the file being played periodically. I do not stream the output so I do not know if the audio also disappears from a stream, I only know the audio monitor channel stops working. I play the OBS output to a projector and websockets to also change the scene periodically. Eventually the audio monitor fails but only for one scene/source. If I wait too long I can find multiple scenes I am using require me to reset their Audio Monitoring option to "Monitor Off" and back to "Monitor and Output". Video playback is not affected as far as I know. Each scene also has browser sources and sometimes additional media sources for example a background video source with another media source for audio. The video plays but the audio stops playing to monitor eventually.

**Comment by WizardCM** [comment]
Please check the log file from the session, and upload it. My guess is the audio buffering hits maximum, which then causes it to silence itself.

Ideally, we'd have a system in place that slowly tries to resync the audio so that it's no longer buffered.

In your case, double check that your audio devices are all set to the same sample rate (in your operating system), and the same goes for OBS - in Settings -> Audio.

**Comment by michaeltoohig** [comment]
I'll have to get back to you on this, perhaps next week, but thanks for the quick reply.

Quick question, does the audio sample rate of the media matter in this case?

**Comment by michaeltoohig** [comment]
Also, (sorry the `Edit` button is not working on my slow internet so here is a follow up post) I'll look into obs-websockets to see if I can change the Audio Monitoring settings. I could live with a periodic task resetting the audio monitoring at midnight each day.

**Comment by WizardCM** [comment]
Ideally, the sample rate on the file shouldn't matter, but make sure it is consistent just in case.

**Comment by michaeltoohig** [comment]
Here is a log file. The last audio settings changed was me modifying the OBS audio settings to 44.8K to match the system settings as you suggested. There is a few audio buffer updates and I see a warning that the maximum buffer was reached.

[log.txt](https://github.com/obsproject/obs-studio/files/4106008/log.txt)

**Comment by michaeltoohig** [comment]
I just tried to `SetSyncOffset` back and forth to 0 to to find if that would reset the buffer but it did not. I also tried toggling mute and again it did not work. I don't see an option in obs-websockets to control the Audio Monitoring.

**Comment by MarioMey** [comment]
Same here. But it is not necessary days of playing to reproduce it. I have a scene with a video with audio, I go in and out this scene and sometimes monitoring works and sometimes it doesn't. The same by entering in "Advanced Audio Properties" and change they 3 monitoring options: sometimes it works, sometimes it doesn't.

I had audio settings in 44.1k and the video was in 48k. After changing settings to 48k, it ocurrs fewer times... but it still ocurrs.

Should I log something? I checked with --verbose but it says nothing different when it sounds and when it doesn't sound. Look at this log, I added when changing options.

```
# monitor off
info: pulse-am: Stopped Monitoring in 'alsa_output.usb-Logitech_Logitech_USB_Headset-00.analog-stereo.monitor'
info: pulse-am: Got 216 packets with 203197 frames
info: User changed audio monitoring for source 'redartistasoeste-01 - video' to: none
# monitor only -> OK
info: pulse-am: Server name: 'pulseaudio 12.2'
info: pulse-am: Audio format: s16le, 44100 Hz, 2 channels
info: pulse-am: Started Monitoring in 'alsa_output.usb-Logitech_Logitech_USB_Headset-00.analog-stereo.monitor'
info: User changed audio monitoring for source 'redartistasoeste-01 - video' to: monitor only
# monitor off
info: pulse-am: Stopped Monitoring in 'alsa_output.usb-Logitech_Logitech_USB_Headset-00.analog-stereo.monitor'
info: pulse-am: Got 97 packets with 91242 frames
info: User changed audio monitoring for source 'redartistasoeste-01 - video' to: none
# monitor only -> OK
info: pulse-am: Server name: 'pulseaudio 12.2'
info: pulse-am: Audio format: s16le, 44100 Hz, 2 channels
info: pulse-am: Started Monitoring in 'alsa_output.usb-Logitech_Logitech_USB_Headset-00.analog-stereo.monitor'
info: User changed audio monitoring for source 'redartistasoeste-01 - video' to: monitor only
# monitor off
info: pulse-am: Stopped Monitoring in 'alsa_output.usb-Logitech_Logitech_USB_Headset-00.analog-stereo.monitor'
info: pulse-am: Got 111 packets with 104413 frames
info: User changed audio monitoring for source 'redartistasoeste-01 - video' to: none
# monitor only -> FAIL
info: pulse-am: Server name: 'pulseaudio 12.2'
info: pulse-am: Audio format: s16le, 44100 Hz, 2 channels
info: pulse-am: Started Monitoring in 'alsa_output.usb-Logitech_Logitech_USB_Headset-00.analog-stereo.monitor'
info: User changed audio monitoring for source 'redartistasoeste-01 - video' to: monitor only

```

**Comment by echorebel** [comment]
I had similar issue today, after a while the audio stop and i get "Max audio buffering reached". I used the setting "Monitoring Off".
Switching Audio Monitoring brings back Audio to the stream. I guess this is expected as the Audio buffer gets reset and reinitialized. Happens to me after random times. Once after 5 minutes, another time after 20 minutes or even ~40 minutes.

I will attach one log file here too:
[2021-02-04 21-34-25.txt](https://github.com/obsproject/obs-studio/files/5928725/2021-02-04.21-34-25.txt)

**Comment by konsolenritter** [comment]
> 
> 
> Here is a log file. The last audio settings changed was me modifying the OBS audio settings to 44.8K to match the system settings as you suggested. There is a few audio buffer updates and I see a warning that the maximum buffer was reached.
> 
> [log.txt](https://github.com/obsproject/obs-studio/files/4106008/log.txt)

The "max audio buffering reached" comes 16 minutes after the last scene switch. Sounds really that your soundcard is a little off. Independent of the actual samplerate your device may be just a tenth of a sample behind (or too fast). Over time that sums up.

To give an anecdote: A couple of months ago we re-recorded a live-streamed online-course with obs on a panasonic toughbook. Interestingly the audio lag of the internal soundcard (it was a little slow in wallclock-time, could have been 44099.4 Hz or something) summed up to more than 30 seconds!! We took the audio from an external pc in by the audio part of an elgato, and played it back thru the internal soundcard of the toughbook. The difference of these two soundcards summed up that way. We saw the speaker ending in the video, but over a couple of hours the audio was that late...! Afterwards we decided to have short breaks all-over the course, and stopped the recording, closed obs-studio for a couple of seconds, reopened it and started recording again, keeping the offset below one second. Crazy, isn't it? We were astonished that obs could have opened up such a lot playback buffers on that windows-10 machine that time.

**Comment by WizardCM** [comment]
Hi @michaeltoohig just wanted to check if you're still having this issue in OBS 27.0.1?

**Comment by michaeltoohig** [comment]
@WizardCM I no longer actively work on that project so that project is still running with OBS version 25. Yet the issue has not appeared in a long time even though that OBS runs 24/7 for months at a time. I can't say what changed except when I started the project I know I had started with OBS version 22 or 23 and was updating as they releases came. Somewhere in there the issue stopped occurring I guess.

Sorry I can't be more helpful than that.

**Comment by michaeltoohig** [comment]
Sorry to others if I preemptively closed the issue if you are still facing it. If so please reopen it.

**Comment by MarioMey** [comment]
I commented above because I had the same/similar problem. But it was in june the 5th of last year.

I just tested this bug in a clean scene collection (2 scenes, 5 sources) and in my scene collection (31 scenes, 130 sources). I put videos with audio in 2 scenes, turn monitoring on and change from scene to scene several times. They always played the audio correctly.

From my comment last year to now, I bougth a new computer... but there has been a lot of OBS updates. So... maybe, as you said "somewhere in there the issue stoppedo ocurring" you guess... so do I.

**Comment by MarioMey** [comment]
I just open a project with VLC source playing some audio files. Audio mixer vumeters showed that audio was playing **but there was no audio in monitor speakers**. I entered in `Advanced Audio Properties` and changed `Audio monitoring` to `Monitor off` and then, again to `Monitor and Output`. Just in this moment, I could hear audio monitoring.

Then, I closed OBS and tried it again... with no success: **audio was always played** 🤦.

What should I do to debug it? I think there is a bug there yet... but I don't know how to reproduce it.
